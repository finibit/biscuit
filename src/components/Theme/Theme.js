import React from 'react'
import PropTypes from 'prop-types'
import { ThemeProvider, createGlobalStyle } from 'styled-components'
import { mergeTheme, baseTheme, themeGet } from '../../themes'
import types from '../../types'

const GlobalStyle = createGlobalStyle`
	html, body {
		width: 100%;
		height: 100%;
		margin: 0;
		padding: 0;
	}

	body {
		color: ${themeGet.color('black', 'inherit')};
		font-family: ${themeGet.fontFamily('secondary')};
		-webkit-font-smoothing: antialiased;
	}
`

const Theme = (props) => {
	const {
		theme,
		global,
	} = props

	return (
		<ThemeProvider theme={mergeTheme(baseTheme, theme)}>
			<div>
				{ global && <GlobalStyle /> }
				{props.children}
			</div>
		</ThemeProvider>
	)
}

Theme.propTypes = {
	global: PropTypes.bool,
	...types.theme,
}

Theme.defaultProps = {
	global: true,
	theme: baseTheme,
}

export default Theme
