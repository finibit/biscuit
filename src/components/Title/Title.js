import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { themeGet } from '@styled-system/theme-get'
import styles from '../../styles'
import types from '../../types'

const TitleStyled = styled.h1`
	box-sizing: border-box;
	font-family: ${themeGet('title.font', themeGet('fonts.primary', 'sans-serif'))};
	font-weight: ${themeGet('title.fontWeight', themeGet('fontWeights.semi-bold', 700))};
	font-size: ${props => themeGet(`fontSizes.${7 - props.size}`, `${props.size}em`)};
	line-height: ${props => props.nowrap ? themeGet(`lineHeight.lg`, '1.5em') : themeGet(`lineHeights.sm`, '1.3em')};
	padding: 0;
	margin: 0 0 ${themeGet('title.margin', themeGet('space.2'), '0.75rem')};

	${styles.compose(
		styles.color,
		styles.text,
	)};

	${styles.nowrap};
`

const Title = (props) => (
	<TitleStyled
		{...props}
	/>
)

Title.propTypes = {
	size: PropTypes.oneOf([1, 2, 3, 4, 5, 6]).isRequired,
	...types.base,
	...types.color,
	...types.text,
	...types.nowrap,
}

Title.defaultProps = {
	color: 'black',
	textAlign: 'left',
	textTransform: 'none',
	textDecoration: 'none',
	nowrap: false,
}

export default Title
