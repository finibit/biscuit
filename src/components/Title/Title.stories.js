import React from 'react'
import { storiesOf } from '@storybook/react'
import Wrapper from '../../Wrapper'
import Title from './Title'
import Box from '../Box'

storiesOf('Title', module)
  .addDecorator(Wrapper)
  .add('with level', () => (
    <>
      <Title level={1}>Title 1</Title>
      <Title level={2}>Title 2</Title>
      <Title level={3}>Title 3</Title>
      <Title level={4}>Title 4</Title>
      <Title level={5}>Title 5</Title>
      <Title level={6}>Title 6</Title>
    </>
  ))
  .add('with color', () => (
    <>
      <Title level={3} color='black'>Black</Title>
      <Title level={3} color='white' bg='black'>White</Title>
      <Title level={3} color='light'>Light</Title>
      <Title level={3} color='primary'>Primary</Title>
      <Title level={3} color='info'>Info</Title>
      <Title level={3} color='danger'>Danger</Title>
      <Title level={3} color='success'>Success</Title>
      <Title level={3} color='warning'>Warning</Title>
    </>
  ))
  .add('with decoration', () => (
    <>
      <Title level={3} textDecoration='overline'>Overline</Title>
      <Title level={3} textDecoration='underline'>Underline</Title>
      <Title level={3} textDecoration='line-through'>Through</Title>
    </>
  ))
  .add('transform', () => (
    <>
      <Title level={3} textTransform='uppercase'>uppercase</Title>
      <Title level={3} textTransform='lowercase'>LOWERCASE</Title>
      <Title level={3} textTransform='capitalize'>capitalize</Title>
      <Title level={3} textTransform='full-width'>Full-width</Title>
    </>
  ))
  .add('align', () => (
    <>
      <Title level={3} textAlign='left'>This title is aligned to the left</Title>
      <Title level={3} textAlign='right'>This is aligned to the right</Title>
      <Title level={3} textAlign='center'>This one is centered</Title>
    </>
  ))
  .add('nowrap', () => (
    <Box width='8rem'>
      <Title level={3} nowrap>Very long title</Title>
    </Box>
  ))
