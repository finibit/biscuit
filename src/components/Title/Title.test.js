import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import Theme from '../Theme'
import Title from './Title'

describe('Title', () => {
  test('with level', () => {
    const component = renderer.create((
      <Theme>
        <Title level={1} />
        <Title level={2} />
        <Title level={3} />
        <Title level={4} />
        <Title level={5} />
        <Title level={6} />
      </Theme>
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })

  test('with color', () => {
    const component = renderer.create((
      <Theme>
        <Title level={1} color='black' />
        <Title level={1} color='white' />
        <Title level={1} color='light' />
        <Title level={1} color='primary' />
        <Title level={1} color='info' />
        <Title level={1} color='danger' />
        <Title level={1} color='success' />
        <Title level={1} color='warning' />
        <Title level={1} color='#001122' />
        <Title level={1} color='cyan' />
      </Theme>
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })

  test('with decoration', () => {
    const component = renderer.create((
      <Theme>
        <Title level={1} textDecoration='overline' />
        <Title level={1} textDecoration='underline' />
        <Title level={1} textDecoration='line-through' />
      </Theme>
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })

  test('transform', () => {
    const component = renderer.create((
      <Theme>
        <Title level={1} textTransform='uppercase' />
        <Title level={1} textTransform='lowercase' />
        <Title level={1} textTransform='capitalize' />
        <Title level={1} textTransform='full-width' />
      </Theme>
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })

  test('align', () => {
    const component = renderer.create((
      <Theme>
        <Title level={1} textAlign='left' />
        <Title level={1} textAlign='right' />
        <Title level={1} textAlign='center' />
      </Theme>
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })

  test('nowrap', () => {
    const component = renderer.create((
      <Theme>
        <Title level={3} nowrap />
      </Theme>
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })
})
