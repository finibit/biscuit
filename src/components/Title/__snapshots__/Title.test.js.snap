// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title align 1`] = `
.c0 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 600;
  font-size: 2.488em;
  line-height: 1.125;
  padding: 0;
  margin: 0 0 16px;
  text-align: left;
  color: #1e272e;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 600;
  font-size: 2.488em;
  line-height: 1.125;
  padding: 0;
  margin: 0 0 16px;
  text-align: right;
  color: #1e272e;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 600;
  font-size: 2.488em;
  line-height: 1.125;
  padding: 0;
  margin: 0 0 16px;
  text-align: center;
  color: #1e272e;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div>
  <h1
    className="c0"
    color="black"
    textDecoration="none"
  />
  <h1
    className="c1"
    color="black"
    textDecoration="none"
  />
  <h1
    className="c2"
    color="black"
    textDecoration="none"
  />
</div>
`;

exports[`Title nowrap 1`] = `
.c0 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 600;
  font-size: 1.728em;
  line-height: 1.5em;
  padding: 0;
  margin: 0 0 16px;
  color: #1e272e;
  text-align: left;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

<div>
  <h1
    className="c0"
    color="black"
    textDecoration="none"
  />
</div>
`;

exports[`Title transform 1`] = `
.c0 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 600;
  font-size: 2.488em;
  line-height: 1.125;
  padding: 0;
  margin: 0 0 16px;
  text-transform: uppercase;
  color: #1e272e;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 600;
  font-size: 2.488em;
  line-height: 1.125;
  padding: 0;
  margin: 0 0 16px;
  text-transform: lowercase;
  color: #1e272e;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 600;
  font-size: 2.488em;
  line-height: 1.125;
  padding: 0;
  margin: 0 0 16px;
  text-transform: capitalize;
  color: #1e272e;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c3 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 600;
  font-size: 2.488em;
  line-height: 1.125;
  padding: 0;
  margin: 0 0 16px;
  text-transform: full-width;
  color: #1e272e;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div>
  <h1
    className="c0"
    color="black"
    textDecoration="none"
  />
  <h1
    className="c1"
    color="black"
    textDecoration="none"
  />
  <h1
    className="c2"
    color="black"
    textDecoration="none"
  />
  <h1
    className="c3"
    color="black"
    textDecoration="none"
  />
</div>
`;

exports[`Title with color 1`] = `
.c0 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 600;
  font-size: 2.488em;
  line-height: 1.125;
  padding: 0;
  margin: 0 0 16px;
  color: #1e272e;
  text-align: left;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 600;
  font-size: 2.488em;
  line-height: 1.125;
  padding: 0;
  margin: 0 0 16px;
  color: #ffffff;
  text-align: left;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 600;
  font-size: 2.488em;
  line-height: 1.125;
  padding: 0;
  margin: 0 0 16px;
  color: #f1f2f6;
  text-align: left;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c3 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 600;
  font-size: 2.488em;
  line-height: 1.125;
  padding: 0;
  margin: 0 0 16px;
  color: #ffdd59;
  text-align: left;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c4 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 600;
  font-size: 2.488em;
  line-height: 1.125;
  padding: 0;
  margin: 0 0 16px;
  color: #0fbcf9;
  text-align: left;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 600;
  font-size: 2.488em;
  line-height: 1.125;
  padding: 0;
  margin: 0 0 16px;
  color: #ff3f34;
  text-align: left;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c6 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 600;
  font-size: 2.488em;
  line-height: 1.125;
  padding: 0;
  margin: 0 0 16px;
  color: #05c46b;
  text-align: left;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c7 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 600;
  font-size: 2.488em;
  line-height: 1.125;
  padding: 0;
  margin: 0 0 16px;
  color: #ffa801;
  text-align: left;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c8 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 600;
  font-size: 2.488em;
  line-height: 1.125;
  padding: 0;
  margin: 0 0 16px;
  color: #001122;
  text-align: left;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c9 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 600;
  font-size: 2.488em;
  line-height: 1.125;
  padding: 0;
  margin: 0 0 16px;
  color: cyan;
  text-align: left;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div>
  <h1
    className="c0"
    color="black"
    textDecoration="none"
  />
  <h1
    className="c1"
    color="white"
    textDecoration="none"
  />
  <h1
    className="c2"
    color="light"
    textDecoration="none"
  />
  <h1
    className="c3"
    color="primary"
    textDecoration="none"
  />
  <h1
    className="c4"
    color="info"
    textDecoration="none"
  />
  <h1
    className="c5"
    color="danger"
    textDecoration="none"
  />
  <h1
    className="c6"
    color="success"
    textDecoration="none"
  />
  <h1
    className="c7"
    color="warning"
    textDecoration="none"
  />
  <h1
    className="c8"
    color="#001122"
    textDecoration="none"
  />
  <h1
    className="c9"
    color="cyan"
    textDecoration="none"
  />
</div>
`;

exports[`Title with decoration 1`] = `
.c0 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 600;
  font-size: 2.488em;
  line-height: 1.125;
  padding: 0;
  margin: 0 0 16px;
  -webkit-text-decoration: overline;
  text-decoration: overline;
  color: #1e272e;
  text-align: left;
  text-transform: none;
}

.c1 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 600;
  font-size: 2.488em;
  line-height: 1.125;
  padding: 0;
  margin: 0 0 16px;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  color: #1e272e;
  text-align: left;
  text-transform: none;
}

.c2 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 600;
  font-size: 2.488em;
  line-height: 1.125;
  padding: 0;
  margin: 0 0 16px;
  -webkit-text-decoration: line-through;
  text-decoration: line-through;
  color: #1e272e;
  text-align: left;
  text-transform: none;
}

<div>
  <h1
    className="c0"
    color="black"
    textDecoration="overline"
  />
  <h1
    className="c1"
    color="black"
    textDecoration="underline"
  />
  <h1
    className="c2"
    color="black"
    textDecoration="line-through"
  />
</div>
`;

exports[`Title with level 1`] = `
.c0 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 600;
  font-size: 2.488em;
  line-height: 1.125;
  padding: 0;
  margin: 0 0 16px;
  color: #1e272e;
  text-align: left;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 600;
  font-size: 2.074em;
  line-height: 1.125;
  padding: 0;
  margin: 0 0 16px;
  color: #1e272e;
  text-align: left;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 600;
  font-size: 1.728em;
  line-height: 1.125;
  padding: 0;
  margin: 0 0 16px;
  color: #1e272e;
  text-align: left;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c3 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 600;
  font-size: 1.44em;
  line-height: 1.125;
  padding: 0;
  margin: 0 0 16px;
  color: #1e272e;
  text-align: left;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c4 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 600;
  font-size: 1.2em;
  line-height: 1.125;
  padding: 0;
  margin: 0 0 16px;
  color: #1e272e;
  text-align: left;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 600;
  font-size: 1rem;
  line-height: 1.125;
  padding: 0;
  margin: 0 0 16px;
  color: #1e272e;
  text-align: left;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div>
  <h1
    className="c0"
    color="black"
    textDecoration="none"
  />
  <h1
    className="c1"
    color="black"
    textDecoration="none"
  />
  <h1
    className="c2"
    color="black"
    textDecoration="none"
  />
  <h1
    className="c3"
    color="black"
    textDecoration="none"
  />
  <h1
    className="c4"
    color="black"
    textDecoration="none"
  />
  <h1
    className="c5"
    color="black"
    textDecoration="none"
  />
</div>
`;
